### PR TITLE
pythonPackages: migrate away from ffmpeg_3

### PR DESCRIPTION
**Pull request by dotlambda**
###### Motivation for this change
https://github.com/NixOS/nixpkgs/issues/120705

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by SuperSandro2000 — COMMENTED** [review]
pkg_resources should be easily fixable by adding a setuptools to propagatedBuildInputs.

**Comment by dotlambda** [comment]
> pkg_resources should be easily fixable by adding a setuptools to propagatedBuildInputs.

The package is python2-only. Why keep a zombie?

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 121257` at 0dcc1e62 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>python38Packages.rl-coach</li>
    <li>python39Packages.rl-coach</li>
  </ul>
</details>
<details>
  <summary>17 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/e48ec6bb4a8071108c097dd102e3b347">python38Packages.aplpy</a></li>
    <li><a href="https://gist.github.com/88775b3a096b796e6151a089ae96657f">python38Packages.asdf</a></li>
    <li><a href="https://gist.github.com/1a02847d01d986c7c271be892e513a71">python38Packages.dask-image</a></li>
    <li><a href="https://gist.github.com/b1d1234fee3a815316b7f685eba79981">python38Packages.pyregion</a></li>
    <li><a href="https://gist.github.com/77fbf7a632e563081b5dcc9b33c329a2">python38Packages.pytest-astropy</a></li>
    <li><a href="https://gist.github.com/60e30885108bf395cc15417afddf8cf8">python38Packages.radio_beam</a></li>
    <li><a href="https://gist.github.com/30a9eacac411aa058e9096f14b64b31a">python38Packages.reproject</a></li>
    <li><a href="https://gist.github.com/8b176bbda8d170a14dcc72d4ac10c54f">python38Packages.runway-python</a></li>
    <li><a href="https://gist.github.com/280088ef9c651d785d51d5ce26ced39d">python38Packages.sunpy</a></li>
    <li><a href="https://gist.github.com/419a742ea1f45ee78c3b236b6645800f">python39Packages.aplpy</a></li>
    <li><a href="https://gist.github.com/1e4af7fa2d21f95fc7695540cd35a6b7">python39Packages.asdf</a></li>
    <li><a href="https://gist.github.com/a3c2923023f08e7b9d032107bf071e0f">python39Packages.dask-image</a></li>
    <li><a href="https://gist.github.com/66465e32958d510c8cbf3698c574d55e">python39Packages.pyregion</a></li>
    <li><a href="https://gist.github.com/b81f0a4f6ccf6875c650f52c2206d2bb">python39Packages.pytest-astropy</a></li>
    <li><a href="https://gist.github.com/8843d0aaf91cef2a98655b2da2c3fbca">python39Packages.radio_beam</a></li>
    <li><a href="https://gist.github.com/23bd5b0cf9b6590ba1ae268a0e745065">python39Packages.reproject</a></li>
    <li><a href="https://gist.github.com/d079d4cf7d0e084099009d554851275a">python39Packages.runway-python</a></li>
  </ul>
</details>
<details>
  <summary>12 packages skipped due to time constraints:</summary>
  <ul>
    <li>python38Packages.caffe</li>
    <li>python38Packages.imgaug</li>
    <li>python38Packages.mask-rcnn</li>
    <li>python38Packages.spectral-cube</li>
    <li>python38Packages.test-tube</li>
    <li>python39Packages.caffe</li>
    <li>python39Packages.imgaug</li>
    <li>python39Packages.mask-rcnn</li>
    <li>python39Packages.moviepy</li>
    <li>python39Packages.spectral-cube</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>25 packages built successfully:</summary>
  <ul>
    <li>python38Packages.batchgenerators</li>
    <li>python38Packages.glymur</li>
    <li>python38Packages.handout</li>
    <li>python38Packages.image-match</li>
    <li>python38Packages.imagecorruptions</li>
    <li>python38Packages.imageio</li>
    <li>python38Packages.imageio-ffmpeg</li>
    <li>python38Packages.moviepy</li>
    <li>python38Packages.pims</li>
    <li>python38Packages.pytest-astropy-header</li>
    <li>python38Packages.scikitimage</li>
    <li>python38Packages.slicedimage</li>
    <li>python38Packages.stytra</li>
    <li>python39Packages.batchgenerators</li>
    <li>python39Packages.glymur</li>
    <li>python39Packages.handout</li>
    <li>python39Packages.image-match</li>
    <li>python39Packages.imagecorruptions</li>
    <li>python39Packages.imageio</li>
    <li>python39Packages.imageio-ffmpeg</li>
    <li>python39Packages.pims</li>
    <li>python39Packages.pytest-astropy-header</li>
    <li>python39Packages.scikitimage</li>
    <li>python39Packages.slicedimage</li>
    <li>tartube</li>
  </ul>
</details>
<details>
  <summary>3 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/development/python-modules/imageio-ffmpeg/default.nix:22:5:
```
   |
22 |     (substituteAll {
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl3` is a deprecated license, please check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/development/python-modules/stytra/default.nix:80:5:
```
   |
80 |     license = lib.licenses.gpl3;
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/imageio/default.nix:39:3:
```
   |
39 |   meta = with lib; {
   |   ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement.

File: pkgs/development/python-modules/imageio-ffmpeg/default.nix [hunk 1]
```diff
@@ -1,29 +1,38 @@
...
 
-  disabled = !isPy3k;
+  patches = [
+    (substituteAll {
+      src = ./ffmpeg-path.patch;
+      ffmpeg = "${ffmpeg}/bin/ffmpeg";
+    })
+  ];
+
+  checkPhase = ''
```
Comment (lukegb) [inline pkgs/development/python-modules/imageio-ffmpeg/default.nix:28]: Should probably run the preCheck and postCheck hooks.

Comment (dotlambda) [reply pkgs/development/python-modules/imageio-ffmpeg/default.nix:28]: I never got why, but sure.

**Pull request merged**